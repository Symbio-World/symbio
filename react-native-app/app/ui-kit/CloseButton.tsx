import * as React from 'react'

import { RoundedButton } from './RoundedButton'
import { ButtonStyle } from './Button'
import { Icons } from './Icon'

type CloseButtonProps = {
  onClose?: () => void
}

export const CloseButton: React.FC<CloseButtonProps> = ({ onClose }) => (
  <RoundedButton
    onPress={onClose}
    outerTitle="close"
    variant={ButtonStyle.OUTLINE}
    icon={Icons.CROSS}
  />
)
