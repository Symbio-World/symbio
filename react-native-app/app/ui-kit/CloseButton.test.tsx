import * as React from 'react'
import { render } from 'react-native-testing-library'
import { CloseButton } from './CloseButton'

describe('CloseButton', () => {
  it('renders correctly', () => {
    const handlePress = jest.fn()
    const { toJSON } = render(<CloseButton onPress={handlePress} />)
    expect(toJSON()).toMatchSnapshot()
  })
})
