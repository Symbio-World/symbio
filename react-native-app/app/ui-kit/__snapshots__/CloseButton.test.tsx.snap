// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CloseButton renders correctly 1`] = `
<View
  accessible={true}
  focusable={true}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "padding": 8,
        },
        Object {
          "backgroundColor": "#fff",
        },
        Object {
          "borderRadius": 8,
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#fff",
          },
          Object {
            "borderWidth": 2,
          },
          Object {
            "borderColor": "#B35340",
          },
          Object {
            "alignItems": "center",
          },
          Object {
            "justifyContent": "center",
          },
          Object {
            "width": 48,
          },
          Object {
            "height": 48,
          },
          Object {
            "borderRadius": 9999,
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "width": 20,
            },
            Object {
              "height": 20,
            },
          ]
        }
      >
        <SvgMock
          height="100%"
          width="100%"
        />
      </View>
    </View>
    <View>
      <Text
        style={
          Array [
            Object {
              "textAlign": "center",
            },
            Object {
              "fontSize": 16,
            },
            Object {
              "color": "#B35340",
            },
            Object {
              "fontWeight": "bold",
            },
          ]
        }
      >
        close
      </Text>
    </View>
  </View>
</View>
`;
